app/controllers/items_controller.rb

```diff
@@ -0,0 +1,47 @@
+class ItemsController < ApplicationController
+  def index
+    @merchant = Merchant.find(params[:merchant_id])
+    @items = @merchant.items
+  end
+
+  def show
+    @merchant = Merchant.find(params[:merchant_id])
+    @item = Item.find(params[:id])
+  end
+
+  def edit
+    @merchant = Merchant.find(params[:merchant_id])
+    @item = Item.find(params[:id])
+  end
+
+  def update
+    item = Item.find(params[:id])
+    item.update!(item_params)
+    if item.save
+      if item_params['enabled'] != nil
+        redirect_to merchant_items_path
+      else
+        redirect_to merchant_item_path
+      end
```

Comment:
I'm not sure what this is doing here.  It seems like both happy and sad paths have the same result of getting redirected to the `merchant_items_path`?

Reply:
If the enabled/disabled button is clicked, the user is redirected to the merchant item index page. If the other attributes are updated, the user is redirected to the merchant item show page. I probably don't need the not successfully updated?

Reply:
Then for both updates, I want the flash updated successful message to appear.